src/java/main/org/apache/zookeeper/server/NettyServerCnxnFactory.java

```diff
@@ -162,7 +162,7 @@ public void messageReceived(ChannelHandlerContext ctx, MessageEvent e)
                             + " from " + ctx.getChannel());
                 }
                 NettyServerCnxn cnxn = (NettyServerCnxn)ctx.getAttachment();
-                synchronized(cnxn) {
+                synchronized(cnxn.getRpcLock()) {
                     processMessage(e, cnxn);
```

Comment:
Now one thread can be in processMessage while another thread is getting stats about the connection.  Is that ok?  

Do you know if anything else besides ServerCnxn is synchronizing on its monitor?  Basically I am wondering if there is any other code out there that should now sync on what getRpcLock() returns.